packages/x-data-grid/src/models/props/DataGridProps.ts

```diff
@@ -787,9 +787,10 @@ export interface DataGridPropsWithoutDefaultValue<R extends GridValidRowModel =
    * @template R
    * @param {R} newRow Row object with the new values.
    * @param {R} oldRow Row object with the old values.
+   * @param {GridRowId} rowId The id of the row.
    * @returns {Promise<R> | R} The final values to update the row.
    */
-  processRowUpdate?: (newRow: R, oldRow: R) => Promise<R> | R;
+  processRowUpdate?: (newRow: R, oldRow: R, rowId: GridRowId) => Promise<R> | R;
```

Comment:
Not a huge fan of having 3 flat params with no apparent hierarchy. 
If the method is not super frequently called, maybe putting the id in a `additionalParams` object would make sure that we stick with 3 params even if we have more info to pass to this function in the future.

Reply:
thanks for the suggestion
the method is called on stop editing and copy paste, I think we should implement this change.